packages/a11y-base/src/aria-hidden.js

```diff
@@ -110,7 +103,7 @@ const applyAttributeToOthers = (originalTarget, parentNode, markerName, controlA
     }
 
     elementsToKeep.add(el);
-    keep(el.parentNode);
+    keep(el.parentNode || el.host);
```

Comment:
This change replaces the previous `unwrapHost` logic to ensure shadow host does not get hidden.